relayer/pkg/chainlink/ocr2/medianreport/onchainconfig.go

```diff
@@ -0,0 +1,113 @@
+package medianreport
+
+import (
+	"fmt"
+	"math/big"
+
+	caigotypes "github.com/dontpanicdao/caigo/types"
+	"github.com/smartcontractkit/chainlink-starknet/relayer/pkg/starknet"
+	"github.com/smartcontractkit/libocr/bigbigendian"
+	"github.com/smartcontractkit/libocr/offchainreporting2/reportingplugin/median"
+)
+
+const (
+	OnchainConfigVersion = 1
+	byteWidth            = 32
+	length               = 3 * byteWidth
+)
+
+// report format
+// 32 bytes - version
+// 32 bytes - min
+// 32 bytes - max
+
+type OnchainConfigCodec struct{}
+
+var _ median.OnchainConfigCodec = &OnchainConfigCodec{}
+
+// DecodeToFelts decodes the onchainconfig into felt values (used in config digest hashing)
+func (codec OnchainConfigCodec) DecodeToFelts(b []byte) ([]*big.Int, error) {
+	if len(b) != length {
+		return []*big.Int{}, fmt.Errorf("unexpected length of OnchainConfig, expected %v, got %v", length, len(b))
+	}
+
+	configVersion, err := bigbigendian.DeserializeSigned(byteWidth, b[:32])
+	if err != nil {
+		return []*big.Int{}, fmt.Errorf("unable to decode version: %s", err)
+	}
+	if OnchainConfigVersion != configVersion.Int64() {
+		return []*big.Int{}, fmt.Errorf("unexpected version of OnchainConfig, expected %v, got %v", OnchainConfigVersion, configVersion.Int64())
+	}
+
+	min, err := bigbigendian.DeserializeSigned(byteWidth, b[byteWidth:2*byteWidth])
+	if err != nil {
+		return []*big.Int{}, err
+	}
+	max, err := bigbigendian.DeserializeSigned(byteWidth, b[2*byteWidth:])
+	if err != nil {
+		return []*big.Int{}, err
+	}
+
+	// ensure felts (used in config digester)
+	min = starknet.BigIntToFelt(min)
+	max = starknet.BigIntToFelt(max)
+
+	return []*big.Int{configVersion, min, max}, nil
+}
+
+// Decode converts the onchainconfig via the outputs of DecodeToFelts into signed big.Ints that libocr expects
+func (codec OnchainConfigCodec) Decode(b []byte) (median.OnchainConfig, error) {
+	felts, err := codec.DecodeToFelts(b)
+	if err != nil {
+		return median.OnchainConfig{}, err
+	}
+
+	// convert felts to big.Ints
+	min := starknet.FeltToBigInt(&caigotypes.Felt{Int: felts[1]})
+	max := starknet.FeltToBigInt(&caigotypes.Felt{Int: felts[2]})
+
+	if !(min.Cmp(max) <= 0) {
+		return median.OnchainConfig{}, fmt.Errorf("OnchainConfig min (%v) should not be greater than max(%v)", min, max)
```

Comment:
```suggestion
		return median.OnchainConfig{}, fmt.Errorf("OnchainConfig min (%v) should be less than max(%v)", min, max)
```

Reply:
the error msg just follows what is used in `libocr` so i'm more inclined to leave it as is for now: https://github.com/smartcontractkit/libocr/blob/db92a9fdaa520171ae8973e5462136402c795151/offchainreporting2/reportingplugin/median/median.go#L64